### PR TITLE
feat: fetch a specific schema version from the api via commit/actionId

### PR DESCRIPTION
**Pull request by n1ru4l**
### Background

See https://github.com/kamilkisiela/graphql-hive/issues/2716

### Description

Closes https://github.com/kamilkisiela/graphql-hive/issues/2716
Closes https://github.com/kamilkisiela/graphql-hive/issues/757

### Checklist

- [ ] Input validation
- [ ] Output encoding
- [ ] Authentication management
- [ ] Session management
- [ ] Access control
- [ ] Cryptographic practices
- [ ] Error handling and logging
- [ ] Data protection
- [ ] Communication security
- [ ] System configuration
- [ ] Database security
- [ ] File management
- [ ] Memory management
- [ ] Testing

### CODE REVIEW

File: packages/libraries/cli/src/commands/schema/fetch.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      aliases: ['W'],
+      description: 'Write to a file (possible extensions: .graphql, .gql, .gqls, .graphqls)',
+    }),
+    outputFile: Flags.string({
+      description: 'whether to write to a file instead of stdout',
+    }),
+  };
+
+  static args = {
+    service: Args.string({
```
Comment (kamilkisiela) [inline packages/libraries/cli/src/commands/schema/fetch.ts:61]: Maybe `actionId`?

File: packages/web/docs/src/pages/docs/features/schema-registry.mdx [hunk 2]
```diff
@@ -174,7 +174,19 @@ performed to ensure a schema is valid:
...
-- [`schema:check` command reference](/docs/api-reference/cli#hive-schemacheck-file)
+
+### Fetch a schema
+
+Sometimes it is useful to fetch a schema (SDL or Supergraph) from Hive, for example, to use it in a
+local development. This can be done using the `schema:fetch` command.
+
+<Callout type="info">
+  Don't confuse this with the [high-availability CDN](/docs/features/high-availability-cdn.mdx).
+  This command is used to fetch a schema from the API wherhe the CDN always represents the latest
```
Comment (kamilkisiela) [inline packages/web/docs/src/pages/docs/features/schema-registry.mdx:185]:
```suggestion
  This command is used to fetch a schema from the API where the CDN always represents the latest
```

File: packages/services/api/src/modules/schema/providers/schema-manager.ts [hunk 3]
```diff
@@ -775,6 +779,36 @@ export class SchemaManager {
...
+      this.organizationManager.getOrganizationFromToken(),
+    ]);
+
+    this.logger.debug('Fetch schema version by action id. (args=%o)', {
+      projectId: target.projectId,
+      targetId: target.id,
+      actionId: args.actionId,
+    });
+
+    const record = await this.storage.getSchemaVersionByActionId({
```
Comment (kamilkisiela) [inline packages/services/api/src/modules/schema/providers/schema-manager.ts:795]: Maybe let's check if the provided token has access to `registry:read-write`?

File: packages/libraries/cli/src/commands/schema/fetch.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      key: 'registry.accessToken',
+      args: flags,
+      env: 'HIVE_TOKEN',
+    });
+
+    const actionId: string = args.service;
+
+    const sdlType = this.ensure({
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```
Comment (B2o5T) [inline packages/libraries/cli/src/commands/schema/fetch.ts:88]:
```suggestion
      // @ts-expect-error $some_comment$
```

**Pull request merged**